src/core/Entry.cpp

```diff
@@ -1373,7 +1378,38 @@ QString Entry::resolveUrl(const QString& url) const
     }
 
     // No valid http URL's found
-    return QString("");
+    return {};
+}
+
+Group* Entry::previousParentGroup()
+{
+    if (!database() || !database()->rootGroup()) {
+        return nullptr;
+    }
+    return database()->rootGroup()->findGroupByUuid(m_data.previousParentGroupUuid);
+}
+
+const Group* Entry::previousParentGroup() const
+{
+    if (!database() || !database()->rootGroup()) {
+        return nullptr;
+    }
+    return database()->rootGroup()->findGroupByUuid(m_data.previousParentGroupUuid);
+}
```

Comment:
No, these are two different code paths, because all the find functions have a cost and a non-const version (which we need both) and the KDBX writer is all const, so I couldn't call any of the non-const members from there. Most of these functions have two implementations just for this. I tried unwinding the const path to avoid duplication, but ran against a wall.